src/liquid/components/ld-file-upload/ld-file-upload.tsx

```diff
@@ -0,0 +1,1034 @@
+  clearChosenFiles() {
+    this.allChosenFiles = []
+  }
+
+  @Listen('ldchoosefiles')
+  updateComponentAppearance() {
+    if (this.renderOnlyChooseFile) {
+      this.renderOnlyChooseFile = false
+    }
+    if (
+      this.circularProgress &&
+      !this.renderCircularProgress &&
+      this.exceedMaxSize.length == 0
+    ) {
+      this.renderCircularProgress = true
+    }
+    // clears file input value
+    this.fileInput.value = null
+  }
+
+  /**
+   * Accepts a file list from component consumer (name, progress, state etc.)
+   * and updates the upload items state.
+   */
+  @Method()
+  async updateUploadItems(uploadItems: UploadItem[]) {
+    if (this.uploadItems.length == 0) {
+      this.uploadItems = uploadItems
+    } else {
+      this.uploadItems = [...this.uploadItems.slice(), ...uploadItems.slice()]
+    }
+  }
+
+  /**
+   * Accepts a file from component consumer (name, progress, state etc.)
+   * and updates the upload item state.
+   */
+  @Method()
+  async updateUploadItem(uploadItem: UploadItem) {
+    const itemToUpdateIndex = this.uploadItems.findIndex(
+      (item) => item.fileName === uploadItem.fileName
+    )
+    if (!itemToUpdateIndex && itemToUpdateIndex != 0) {
+      throw new Error(
+        `Upload item with name ${uploadItem.fileName} not found in upload list.`
+      )
+    }
+
+    this.uploadItems = [
+      ...this.uploadItems.slice(0, itemToUpdateIndex),
+      uploadItem,
+      ...this.uploadItems.slice(itemToUpdateIndex + 1),
+    ]
+  }
+
+  /**
+   * Accepts a file list from component consumer (name, progress, state etc.)
+   * and deletes the upload items.
+   */
+  @Method()
+  async deleteUploadItems() {
+    /* for (let i = 0; i < uploadItems.length; i++) {
+      const itemToDeleteIndex = this.uploadItems.findIndex(
+        (item) => item.fileName === uploadItems[i].fileName
+      )
+      if (!itemToDeleteIndex && itemToDeleteIndex != 0) {
+        throw new Error(
+          `Upload item with name ${uploadItems[i].fileName} not found in upload list.`
+        )
+      }
+
+      this.uploadItems = [
+        ...this.uploadItems.slice(0, itemToDeleteIndex),
+        ...this.uploadItems.slice(itemToDeleteIndex + 1),
+      ]
+    } */
+    this.uploadItems = []
+  }
+
+  /**
+   * Accepts a file from component consumer (name, progress, state etc.)
+   * and deletes the upload items.
+   */
+  @Method()
+  async deleteUploadItem(uploadItem: UploadItem) {
+    const itemToDeleteIndex = this.uploadItems.findIndex(
+      (item) => item.fileName === uploadItem.fileName
+    )
+    if (!itemToDeleteIndex && itemToDeleteIndex != 0) {
+      throw new Error(
+        `Upload item with name ${uploadItem.fileName} not found in upload list.`
+      )
+    }
+
+    this.uploadItems = [
+      ...this.uploadItems.slice(0, itemToDeleteIndex),
+      ...this.uploadItems.slice(itemToDeleteIndex + 1),
+    ]
+  }
+
+  @Watch('uploadItems')
+  async changeProgressVisualisation() {
```

Comment:
Maybe we can refactor this method and extract some of the code in separate functions.